cpu/esp8266/esp/FreeRTOS.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * FreeRTOS to RIOT-OS adaption module
+ */
+
+#ifndef FREERTOS_H
+#define FREERTOS_H
+
+#ifndef DOXYGEN
+
+#include "stdlib.h"
+
+#include "mutex.h"
+#include "irq.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SemaphoreHandle_t mutex_t*
+
+static inline SemaphoreHandle_t xSemaphoreCreateMutex(void)
+{
+    mutex_t* _tmp = malloc (sizeof(mutex_t));
```

Comment:
I think RIOT prefers mutex_t  *_tmp = malloc (sizeof(mutex_t));
At least the uncrustify does...

Reply:
It's just a very old habit :-(

You'll probably find it in many lines of source code from me. I've always used ```type* `` in the past because the star is part of the declaration of the type and not part of the identifier. This makes it easier to read from my point of view.

Reply:
I like it too, it was just a thought, I am not sure how strictly these things are enforced.